### ISSUE TITLE
Operator actions on Interval/Period instances produce timedelta objects

### ISSUE THREAD

**Issue by mayfield** [sdispater/pendulum#26]
Doing various math operator actions on timedelta subclasses produces timedelta instances instances.  This is partially because of the frailty of datetime.timedelta code but it would be highly preferable to support operations like `avg_perdio = total_period / some_count` and get a Period/Interval object back instead of a timedelta object.  It's also quite confusing since the APIs are very different between all 3 of these subclasses.

Example..
```
>>> import pendulum
>>> iobj = pendulum.Interval(seconds=10)
>>> x = iobj / 3
>>> x
datetime.timedelta(0, 3, 333333)
>>> type(x)
<class 'datetime.timedelta'>
>>> type(iobj)
<class 'pendulum.interval.Interval'>
>>> 
```

I took a look at the code and unfortunately it looks like you'd have to subclass all the relevant operator calls (`__div__`, etc) and patch them to produce new Interval or Period objects.  Because Interval and Period have very different instantiation patterns it might not even be possible to implement in the base Interval class. However I'll leave it to you to review.

**Comment by sdispater** [comment]
I think by only overriding the appropriate methods in the `Interval` class but returning new instances using `self.__class__()` we should be ok.

**Comment by sdispater** [comment]
I just tested it and you are right its not that easy. However, I think that arithmetic operations on `Period` instances don't make sense. They represent a duration between two dates so there is no point in multiplying or dividing them. A `Period` inherits from `Interval` for convenience but they are not exactly the same thing.

**Comment by mayfield** [comment]
I don't disagree in terms of type consistency but the distinction between Interval and Period is sometimes just an ancillary effect if you do math with Pendulum objects.  In my case I end up with Period objects because my deltas result from doing things like `thing = pendulum.now() - past_pendulum_obj`.  It's not super important to my use case that I have a Period.

As such, if returning an Interval object in those cases was more logical I'd be good with that.  I just want all those nice methods like `as_words()`.

**Comment by sdispater** [comment]
I see what you mean. I will try to find a way to make it work.

**Comment by sdispater** [comment]
I just pushed a fix (a7ac6fa90a193cb4a7f5a9203beb340bb4a3fa75).

Arithmetic operations on `Interval` instances should now work properly and return an `Interval`.

Regarding `Period` instances, these operations will now return an `Interval`.
